### ISSUE TITLE
[HOLD for payment 2025-01-15]  Advanced approval workflow doesn't work after disabling and re enabling it

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54592]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-2
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://github.com/Expensify/App/pull/52537
**Email or phone of affected tester (no customers):** nathanmulugetatesting+2612@gmail.com
**Issue reported by:** Applause Internal Team
**Device used:** Windows 11 pro/chrome
**App Component:** Money Requests

## Action Performed:
Pre conditions:
- Create a workspace as an owner
- Enable workflows and setup an advanced workflow where submitter submits to approver 1 then approver 1 approves to approver 2

1. Submit an expense as the submitter
2. Sign in as the approver 1 and approve the expense
3. Sign in as the approver 2 and approve the expense
4. As the owner disable approval and re enable it after a few seconds
5. Submit another expense as the submitter

## Expected Result:
The submitted expense appears for approver 1 to approve

## Actual Result:
The submitted expense directly goes to owner to approve and approve 1 is not the one doing the approving first as per the workflow setup

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/5f6eab79-27fe-479b-8d9d-2a988dbc4dd4

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune</details>

**Comment by nkdengineer** [comment]
@Beamanator Is this a BE bug? After we disable and enable approval again the `approvalMode` is `BASIC` not `ADVANCE`

**Comment by Beamanator** [comment]
Innnnnnnteresting... That sounds like something @tgolen has worked on in the backend recently if this has to do with changing the approval mode via NewDot

**Comment by slafortune** [comment]
@tgolen thoughts on this?

**Comment by tgolen** [comment]
I don't think this is a backend issue.

When the workflows are disabled, the FE calls the `SetWorkspaceApprovalMode` command with the value `value: {"approver":"tgolen@expensify.com","approvalMode":"OPTIONAL","isAutoApprovalEnabled":false}`. You can see the approval mode being set to OPTIONAL.

When the workflows are enabled again, the FE calls the `SetWorkspaceApprovalMode` with the value `value: {"approver":"tgolen@expensify.com","approvalMode":"BASIC","isAutoApprovalEnabled":false}`. You can see the approval mode is being set to BASIC.

The frontend should be setting it to ADVANCED.

**Comment by nkdengineer** [comment]
@tgolen We're always passing `BASIC` here when enabling the approval. How can we detect the approval modal should be basic or not? I can work on this after we can clarify a way to detect that.

https://github.com/Expensify/App/blob/5da6a32d33ef040b52994f443440abfcbe3d42fb/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx#L166

**Comment by tgolen** [comment]
I noticed that after I enabled workflows, the UI still showed both workflows that I previously added. So, I assume that the workflow data still exists in Onyx. If that's the case, then you can pass BASIC if there was a single workflow and pass ADVANCED if there were multiple workflows.

**Comment by nkdengineer** [comment]
@tgolen Just tested and it works well. I noticed that we could have another case that is also advance approval mode is only has one workflow but multiple approvers. Here is the solution and I'd like to work on this.

```
const isAdvanceApproval = approvalWorkflows.length > 1 || (approvalWorkflows?.at(0)?.approvers ?? []).length > 1;
const approvalMode = isAdvanceApproval ? CONST.POLICY.APPROVAL_MODE.ADVANCED : CONST.POLICY.APPROVAL_MODE.BASIC;
Policy.setWorkspaceApprovalMode(route.params.policyID, policy?.owner ?? '', isEnabled ? approvalMode : CONST.POLICY.APPROVAL_MODE.OPTIONAL);
```

**Comment by tgolen** [comment]
Nice! That sounds good, and you're right about the mult-approver case.

**Comment by tgolen** [comment]
I'll assign this to you and you can write a PR for it. I'll make sure the payments get sorted out later.

**Comment by ikevin127** [comment]
I was auto-assigned as reviewer, if review is needed I'll start the review in ~10 hours.

**Comment by tgolen** [comment]
Yeah, we should have a C+ review it. Thanks for volunteering @ikevin127 it's all yours!

**Comment by ikevin127** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by ikevin127** [comment]
cc @slafortune for visibility

**Comment by slafortune** [comment]
@ikevin127 offer sent - https://www.upwork.com/nx/wm/offer/105731477
@nkdengineer offer sent - https://www.upwork.com/nx/wm/offer/105731487

**Comment by slafortune** [comment]
All Paid
